src/Raven.Server/Documents/Operations/Operations.cs

```diff
@@ -140,6 +139,7 @@ void ProgressNotification(IOperationProgress progress)
 
             operation.Task = Task.Run(() => taskFactory(ProgressNotification));
 
+            var locker = new ReaderWriterLockSlim();
```

Comment:
why use reader write lock instead of a simple object `lock` ?

Reply:
I don't have to wait until the lock finish so I just exit (line 179)

Reply:
`Monitor.TryEntry` ?

Reply:
done